### ISSUE TITLE
Very frequent Browser black-screen crashes (WEBGL error messages)

### ISSUE THREAD

**Issue by cfravel** [decentraland/unity-renderer#1739]
I am seeing extremely frequent browser black-screen crashes in Decentraland.

I have seen this in many places in DCL, including
* the 0,0 landing spot, 
* the Unti\ty Cafe at 14,100
* In largely empty land such as 150,150, 
* At -12,92 in the road between the University and the Consensys build.
* visiting Vegas
.
I have seen it on 3 different Windows 10 laptops, with NVIDIA:GPUs
* GTX 740
* GTX 1070
* RTX 2070

 (and it occurs with GPU turned off for the browser, too)
 
I have seen it in Chrome, Brave and Edge

It has been getting worse and worse for a few weeks, and today it has gotten so bad that i basically cannot enter most places in DCL in either Chrome or Brave without an immediate crash or a crash within a minute or so.

I have latest NVidia drivers and latest browser versions

Typically what I see in console is
[.WebGL-0000410E0F2CB100] GL_CONTEXT_LOST_KHR: Context has been lost.
or
WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost

This can have some variants, such as:

[.WebGL-00000E9212AE3800] GL_INVALID_OPERATION: Level of detail outside of range.
   And then later either 
CONTEXT_LOST_WEBGL: loseContext: context lost

Or

WebGL-0000410E0F2CB100] GL_OUT_OF_MEMORY: Internal error: 0x00000505: Failed to allocate host memory
94[.WebGL-0000410E0F2CB100] GL_CONTEXT_LOST_KHR: Context has been lost.
/?island=I17cwh&position=-150%2C150&realm=dg:1 WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost
/?island=I17cwh&position=-150%2C150&realm=dg:1 WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost

**Comment by cfravel** [comment]
I think I've found the underlying cause.
If one loads a video texture  in the scene, from a steam that actually isn't running, this seems to be the aftermath.

The error I see in console before the GL_OUT_OF_MEMORY is similar to:

rollbar.min.js:1 GET https://example.com/live/index.m3u8 404 (Not Found)

(where example.com is a node media server forwarding an RTMP via hls with CORS, but the source stream is not running)

The remaining question, if this is the cause, is why tht error would cause out of memory error immediately after.
Does there need to be some error handler in the sdk?   Or is there a way I can handle the error in my scene that would keep it from running out of WEBGL memory?

**Comment by cfravel** [comment]
My video stream thought above isn't conclusive.

I saw the same crash (GL_CONTEXT_LOST_KHR) when turning around and looking at stuff at 74,74, the plaza corner / entrance to Aetheria.   I cannot say that there is for sure no "dead video stream" near there, but I am not aware of one there.
So it remains a mystery.

**Comment by cfravel** [comment]
two more crashes in the last hour.  Brave.   
play.decentraland.org/:1 WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost
play.decentraland.org/:1 WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost

this is on my machine with an RTX 2070 that's only running at aboug 20% of GPU capacity.

**Comment by cfravel** [comment]
4 crashes in 20 minutes.  This is awful.
JamesA says he sees this, too

**Comment by cfravel** [comment]
Crashed tonight while i was demonstrating to a prospective rental client.
Please fix this asap

**Comment by cfravel** [comment]
Tested today with a high end machine but GPU turned off for browser.   Still happens, so it isn't my GPU drivers.

**Comment by cfravel** [comment]
Crashed in Vegas, too, today

**Comment by cfravel** [comment]
i have not seen one of these crashes now in two days, several hours of being active in Chrome
I have been running in Medium quality, despite being on an RTX-2070 GPU

**Comment by olavra** [comment]
Is this still happening?

**Comment by cfravel** [comment]
Somewhere between Jan 19 and Jan 20 this stopped happening for me.
I've visited Conference Center, WonderZone, Aetheria, Ice Poker, 
I've been in Medium graphics quality on my gaming laptop NVIDIA RTX 2070, scene radius 4 draw distance 150

Switched to Ultra (Draw distance 500. sceme radius 6
Tried reversing my steps, and in WonderZone I got something I'd never seen before
The upper half of the screen filled with a dump, and an unhappy welcome bot face showed up
Seems like something else, i'll record it here, then try some mor in Ultra.

[WonderMine error dump to screen.txt](https://github.com/decentraland/unity-renderer/files/7930835/WonderMine.error.dump.to.screen.txt)

![image](https://user-images.githubusercontent.com/21015643/150917381-5535834c-0b32-40bd-b79d-881e9cbe1b15.png)

**Comment by cfravel** [comment]
After restarting the browser being in Ultra + 6 scene radius, i entered the Conference Center Cafe.
went up to the dj console, pressed a button
Blackscreen crash

WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost
play.decentraland.org/:1 WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost

**Comment by cfravel** [comment]
restarted browser, reentered the cafe, ran around a bit, noticed a lot more yellow WEBGL messages, like LOD.
Then blackscreen crash again

play.decentraland.org/:1 WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost
play.decentraland.org/:1 WebGL: CONTEXT_LOST_WEBGL: loseContext: context lost

**Comment by cfravel** [comment]
Maybe I only see this in ULTRA.

I've been medium and then high for awhile, some of the normal scenes i test with this and no problems

I suggest you try this with ULTRA and a high end graphics GPU.   Mine, again, is an NVIDIA RTX 2070

**Comment by cfravel** [comment]
I've seen it in HIGH occasionally too.
it's a pretty hard crash, it can affect Discord sometimes, causing both to go blank
(when my activity wasn't in discord, but was in DCL)

**Comment by olavra** [comment]
Hi @cfravel!

Is this still happening?

**Comment by cfravel** [comment]
i have run on Ultra and see the same problem
I have purchased a new higher-end gaming laptop (MSI GS-76 with an RTX 3070 GPU.
Set it to ULTRA.
Went into The Gallery at https://play.decentraland.org/?position=9%2C98
Ran to the left stairs, went up the stairs and toward the far East end of the Mezzaine
Blackscreen crash
Restarted, 
Didn't reproduce within a couple of minutes.

So, yes, it is still happening.
But it isn't super easy to reproduce quickly.
